[clang-tidy] new check misc-use-internal-linkage (#90830)
llvm/llvm-project: clang-tools-extra/docs/clang-tidy/checks/readability/unnecessary-external-linkage.rst

.. title:: clang-tidy - readability-unnecessary-external-linkage

readability-unnecessary-external-linkage
========================================

Detects variable and function can be marked as static.

Static functions and variables are scoped to a single file. Marking functions
and variables as static helps to better remove dead code. In addition, it gives

> the compiler more information and can help compiler make more aggressive
> optimizations.

Reviewer: ```suggestion
the compiler more information and allows for more aggressive optimizations.
```
